qiskit/tools/visualization/_bloch.py

```diff
@@ -39,17 +39,32 @@
 #    OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 ###############################################################################
 
+# pylint: disable=ungrouped-imports
 
 """Bloch sphere"""
 
 __all__ = ['Bloch']
 
 import os
+import sys
+
 import numpy as np
-import matplotlib.pyplot as plt
 from matplotlib.patches import FancyArrowPatch
 from mpl_toolkits.mplot3d import (Axes3D, proj3d)
 
+if sys.platform == "darwin":
+    try:
+        from matplotlib import pyplot as plt
+    except ImportError:
+        # NOTE(mtreinish): On OSX if python not installed as a framework the
+        # default OSX interactive framework will not work. In that case use
+        # a different backend.
+        import matplotlib
+        matplotlib.use('PS')
```

Comment:
I know why this is here, but I do not think that we should be implicitly changing the users mpl backend.  It could cause issues with other drawing functionality.

Reply:
Well my theory was that if we can't import pyplot on osx then it's most likely the framework issue. mpl is already implicitly picking a backend by setting a default. If a user wants a specific backend they can specify it before the qiskit import to override this easily enough (if pyplot is importable this never runs).

An alternative we could try is switching to runtime imports of matplotlib for the visualizers that use it. The problem is with this patch the `qiskit.tools.visualization` namespace gets imported on any qiskit import, which means that we're pulling in matplotlib unconditionally now. Which breaks pretty every test on the osx CI because of the framework issue.

Reply:
I will add a patch on top of this adding a release note about the feature, and will put some text in there about this change and try to word it very clearly about this change.